docs/rfcs/env-vars.md

```diff
@@ -0,0 +1,216 @@
+# Stabilizing environment variable resolution
+
+## Overview
+
+The OpenTelemetry Collector supports three different syntaxes for
+environment variable resolution which differ in their syntax, semantics
+and allowed variable names. Before we stabilize confmap, we need to
+address several issues related to environment variables. This document
+describes: the current (as of v0.97.0) behavior of the Collector, the
+goals that an environment variable resolution should aim for, existing
+deviations from these goals and the desired behavior after making some
+changes.
+
+CLI environment variable resolution has a single syntax (`--config env:ENV`) 
+and it is considered out of scope for this document, focusing
+instead on expansion within the Collector configuration.
+
+How to get from the current to desired behavior is also considered out
+of scope and will be discussed on individual PRs. It will likely involve
+one or multiple feature gates, warnings and transition periods.
+
+## Goals of an expansion system
+
+The following are considered goals of the expansion system:
+
+1.  ***Expansion should happen only when the user expects it***. We
+    should aim to expand when the user expects it and keep the original
+    value when we don't (e.g. because the syntax is used for something
+    different).
+2.  ***Expansion should have predictable behavior***.
+3.  ***Multiple expansion methods, if present, should have similar behavior.***
+    Switching from `${env:ENV}` to `${ENV}` or vice versa
+    should not lead to (too many?) surprises.
+4.  ***When the syntax overlaps, expansion should be aligned with*** 
+    [***the expansion defined by the Configuration Working Group***](https://github.com/open-telemetry/opentelemetry-specification/blob/032213cedde54a2171dfbd234a371501a3537919/specification/configuration/file-configuration.md#environment-variable-substitution).
```

Comment:
I filed open-telemetry/opentelemetry-specification/issues/3963 to ensure this. I don't think this should completely block merging this RFC: I think at this point it is unavoidable that there will be differences with respect to the File Configuration SDK spec, but this should be a design goal.